src/main/java/uk/gov/hmcts/reform/em/stitching/domain/validation/CallableEndpointValidator.java

```diff
@@ -0,0 +1,36 @@
+package uk.gov.hmcts.reform.em.stitching.domain.validation;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+public class CallableEndpointValidator implements ConstraintValidator<CallableEndpoint, String> {
+
+    private final Logger log = LoggerFactory.getLogger(CallableEndpointValidator.class);
+
+    @Override
+    public boolean isValid(String url, ConstraintValidatorContext context) {
+
+        boolean valid;
+
+        try {
+            URL siteURL = new URL(url);
+            HttpURLConnection connection = (HttpURLConnection) siteURL.openConnection();
+            connection.setRequestMethod("POST");
+            connection.setConnectTimeout(3000);
+            connection.connect();
+            int responseCode = connection.getResponseCode();
+            connection.disconnect();
+            valid = responseCode < 500;
```

Comment:
Is there zero chance a 4** HTTP error will be returned?

Reply:
400 means a success - it means that a request was incorrect, but still was able to connect

With this Validator all I want to test is whether I can connect to this endpoint (resolve domain, initiate http connection)

Reply:
👍